docs/includes/fundamentals/as-avs/AtlasSearchTest.php

```diff
@@ -0,0 +1,158 @@
+            $moviesCollection->createSearchIndex([
+                'mappings' => [
+                    'fields' => [
+                        'title' => [
+                            ['type' => 'string', 'analyzer' => 'lucene.english'],
+                            ['type' => 'autocomplete', 'analyzer' => 'lucene.english'],
+                            ['type' => 'token'],
+                        ],
+                    ],
+                ],
+            ]);
+
+            $moviesCollection->createSearchIndex([
+                'mappings' => ['dynamic' => true],
+            ], ['name' => 'dynamic_search']);
+
+            $moviesCollection->createSearchIndex([
+                'fields' => [
+                    ['type' => 'vector', 'numDimensions' => 4, 'path' => 'vector4', 'similarity' => 'cosine'],
+                    ['type' => 'filter', 'path' => 'title'],
+                ],
+            ], ['name' => 'vector', 'type' => 'vectorSearch']);
+        } catch (ServerException $e) {
+            if (Builder::isAtlasSearchNotSupportedException($e)) {
+                self::markTestSkipped('Atlas Search not supported. ' . $e->getMessage());
+            }
+
+            throw $e;
+        }
+
+        // Wait for the index to be ready
+        do {
+            $ready = true;
+            usleep(10_000);
+            foreach ($collection->listSearchIndexes() as $index) {
+                if ($index['status'] !== 'READY') {
+                    $ready = false;
+                }
+            }
+        } while (! $ready);
+    }
+
+    /**
+     * @runInSeparateProcess
+     * @preserveGlobalState disabled
+     */
+    public function testSimpleSearch(): void
+    {
+        // start-search-query
+        $movies = Movie::search(
+            sort: ['title' => 1],
+            operator: Search::text('title', 'dream'),
+        )->get();
+        // end-search-query
+
+        $this->assertNotNull($movies);
+        $this->assertCount(2, $movies);
+    }
+
+    /**
+     * @runInSeparateProcess
+     * @preserveGlobalState disabled
+     */
+    public function autocompleteSearchTest(): void
+    {
+        // start-auto-query
+        $movies = Movie::autocomplete('title', 'jak')
+            ->get();
+        // end-auto-query
+
+        $this->assertNotNull($movies);
+        $this->assertCount(3, $movies);
+    }
+
+    /**
+     * @runInSeparateProcess
+     * @preserveGlobalState disabled
+     */
+    public function vectorSearchTest(): void
+    {
+        $results = Book::vectorSearch(
+            index: 'vector',
+            path: 'vector4',
+            queryVector: $this->vectors[0],
+            limit: 3,
+            numCandidates: 10,
+            filter: Query::query(
+                title: Query::ne('A'),
+            ),
+        );
+
+        $this->assertNotNull($results);
+        $this->assertSame('C', $results->first()->title);
+    }
+
+    /** Generate random vectors using fixed seed to make tests deterministic */
```

Comment:
```suggestion
    /** Generates random vectors using fixed seed to make tests deterministic */
```
S: Same guidance as above